api/proto/message/message.proto

```diff
@@ -102,3 +92,18 @@ message DrandRequest {
   bytes block_hash = 2;
   bytes payload = 3;
 }
+
+message ViewChangeRequest {
+    uint32 consensus_id = 1;
+    uint64 seq_num = 2;
+    bytes sender_pubkey = 3;
+    bytes leader_pubkey = 4;
+    bytes payload = 5; // message payload: either m1 type or m2 type 
+    bytes viewchange_sig = 6; // signature on payload
+                        
+    // below is for newview message only
+    bytes m1_aggsigs = 7; // m1: |block_hash|bhp_sigs|bhp_bitmap|
```

Comment:
what is m1 and m2? The <view change> and <new view> message?

Reply:
m1 type message payload records the prepared message a validator received and then start a view change because of timeout.  m2 type message is just the NIL message which means the validator not receive any prepared message yet and start a view change. These are the only two possible two situations.